cardinal/server/server.go

```diff
@@ -130,6 +131,68 @@ func getValueFromParams[T any](params interface{}, name string) (*T, bool) {
 	return value, true
 }
 
+type ProcessedTxResults struct {
+	rawPayload    []byte //could be original request or body of signedPayload
+	signedPayload *sign.SignedPayload
+	transaction   transaction.ITransaction
+}
+
+func processTxParams(
+	params interface{},
+	pathParam string,
+	txNameToTx map[string]transaction.ITransaction,
+	handler *Handler) (*ProcessedTxResults, error) {
+	mappedParams, ok := params.(map[string]interface{})
+	if !ok {
+		return nil, errors.New("params not readable")
+	}
+	var tx transaction.ITransaction
+	if len(pathParam) > 0 {
```

Comment:
You're right, good catch. Should return an error.

Reply:
In that case, I think this should be broken up into "getTxFromParams" and "getBodyAndSigFromParams". the pathParam and txNameToTx would only need to be passed to the getTxFromParams method. Since the input params and returned params would be simplified, the processedTxResults could also be removed. (The first method would just return the tx iface and an error, and the second method would return the body, signed payload, and error).

Reply:
Done